services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -132,7 +132,7 @@ public Response createResponse(UserModel.RequiredAction action) {
                 page = LoginFormsPages.LOGIN_UPDATE_PROFILE;
                 break;
             case UPDATE_PASSWORD:
-                actionMessage = Messages.UPDATE_PASSWORD;
+                actionMessage = user.getRequiredActions().isEmpty() ? Messages.RESET_PASSWORD : Messages.UPDATE_PASSWORD;
```

Comment:
Should it check if user.getRequiredActions contains the update password action rather than checking if it's empty? Thinking there could be cases where a user has some other action set.